src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Middlewares/EvalProductsPricesMiddleware.cs

```diff
@@ -65,9 +68,13 @@ public virtual async Task Run(SearchProductResponse parameter, Func<SearchProduc
                 var productsWithoutPrices = parameter.Results.Where(x => !x.IndexedPrices.Any()).ToArray();
                 if (productsWithoutPrices.Any())
                 {
+                    // find Store by Id to get Catalog Id
+                    var store = await _storeService.GetByIdAsync(query.StoreId, StoreResponseGroup.StoreInfo.ToString());
+
                     var evalContext = AbstractTypeFactory<PricingModule.Core.Model.PriceEvaluationContext>.TryCreateInstance();
                     evalContext.Currency = query.CurrencyCode;
                     evalContext.StoreId = query.StoreId;
+                    evalContext.CatalogId = store?.Catalog;
```

Comment:
Shouldn't it be `query?.Catalog`?

Reply:
there's no `catalogId` in the query, only `storeId`